src/features/search/searchSlice.ts

```diff
@@ -0,0 +1,43 @@
+import { createSelector, createSlice, PayloadAction } from '@reduxjs/toolkit';
+import { RootState } from '@common/state/store';
+
+export interface State {
```

Comment:
Given this is exported and used externally, it'd be better to give it a specific name
```suggestion
export interface SearchState {
```